book/src/todo_3.md

```diff
@@ -0,0 +1,205 @@
++                          <object class="GtkEntry" id="entry">
++                            <property name="placeholder-text" translatable="yes">Enter a Task…</property>
++                            <property name="secondary-icon-name">list-add-symbolic</property>
++                          </object>
++                        </child>
++                        <child>
++                          <object class="GtkListBox" id="tasks_list">
++                            <property name="visible">False</property>
++                            <property name="selection-mode">none</property>
++                            <style>
++                              <class name="boxed-list" />
++                            </style>
++                          </object>
++                        </child>
++                      </object>
++                    </property>
++                  </object>
++                </property>
++              </object>
++            </property>
+           </object>
+         </child>
+       </object>
++    </property>
+   </template>
+ </interface>
+```
+
+We replaced the `gtk::ApplicationWindow` with `adw::ApplicationWindow` and added a `adw::HeaderBar` to it.
+In order to follow the boxed list pattern, we started using [`gtk::ListBox`](../docs/gtk4/struct.ListBox.html), set its property "selection-mode" to "none" and let it match with the `boxed-list` style class. 
+
+Let's continue with `window/imp.rs`.
+The member variable `tasks_list` now describes a `ListBox` rather than a `ListView`.
+We also change the `ParentType` from `gtk::ApplicationWindow` to `adw::ApplicationWindow`.
+
+Filename: <a class=file-link href="https://github.com/gtk-rs/gtk4-rs/blob/master/book/listings/todo/6/window/imp.rs">listings/todo/6/window/imp.rs</a>
+
+```rust,no_run,noplayground
+{{#rustdoc_include ../listings/todo/6/window/imp.rs:window}}
+```
+
+We don't override any function of `adw::ApplicationWindow`, but we still have to add the empty `impl`.
+
+Filename: <a class=file-link href="https://github.com/gtk-rs/gtk4-rs/blob/master/book/listings/todo/6/window/imp.rs">listings/todo/6/window/imp.rs</a>
+
+```rust,no_run,noplayground
+{{#rustdoc_include ../listings/todo/6/window/imp.rs:AdwApplicationWindowImpl}}
+```
+
+We now move on to `window/mod.rs`.
+First, we add `adw::ApplicationWindow` to our list of derived classes.
+
+Filename: <a class=file-link href="https://github.com/gtk-rs/gtk4-rs/blob/master/book/listings/todo/6/window/mod.rs">listings/todo/6/window/mod.rs</a>
+
+```rust,no_run,noplayground
+{{#rustdoc_include ../listings/todo/6/window/mod.rs:glib_wrapper}}
+```
+
+`ListBox` supports models just fine, but without any widget recycling we don't need factories anymore.
+`setup_factory` can therefore be safely deleted.
+To setup the `ListBox`, we call `bind_model` in `setup_tasks`.
+There we specify the model, as well as a closure describing how to transform the given GObject into a widget the list box can display. 
```

Comment:
```suggestion
There we specify the model, as well as a closure describing how to transform the given `GObject` into a widget the list box can display. 
```

Reply:
Leaving as is to stay consistent with other chapters